### ISSUE TITLE
Submit in-memory data as file.

### ISSUE THREAD

**Issue by ioquatix** [rest-client/rest-client#488]
Here is a class which makes that possible, by pretending to be a file in the eyes of RestClient:

``` ruby
class DataFile < StringIO
    def self.from_response(response, filename = 'file')
        self.new(response, response.headers[:content_type], filename)
    end

    def initialize(data, content_type, original_filename)
        @original_filename = original_filename
        @content_type = content_type

        super(data)
    end

    attr :original_filename
    attr :content_type

    alias path original_filename
end
```

**Comment by ioquatix** [comment]
It would be awesome to see this merged into 2.0.0

**Comment by ioquatix** [comment]
Do you use multipart-post gem? I was thinking of adding this to that gem.
